lib/machinery.rb

```diff
@@ -37,6 +37,9 @@
 require "diffy"
 require "sinatra/base"
 require "mimemagic"
+require "docker"
+require "rubygems"
+require "rubygems/package"
```

Comment:
In order to gain access to the `Gem::Package::TarReader` class which we use for unpacking the tarballs.